Themis/REST/RESTController.swift

```diff
@@ -29,6 +29,12 @@ class RESTController {
         let request = try makeURLRequest(request)
         let (data, response) = try await URLSession.shared.data(for: request)
         try validate(response: response)
+
+        // check for empty response
+        guard !data.isEmpty else {
+            throw RESTError.empty
+        }
```

Comment:
Why not put this in a positive `if` instead of the double negation for better readability?
```suggestion
        if data.isEmpty {
            throw RESTError.empty
        }
```

Also, I'm not sure if we should even throw an error for empty responses or rather have a separate function which doesn't require conversion into some object. Throwing because of a perfectly normal HTTP 200 response seems counterintuitive.

Reply:
The guard thing is just a thing of preference. Got used to using it as they always told us that it is more readable in the intro course.

Reply:
Don't really care how we do it

Reply:
Throwing the empty error is unfortunately pretty much necessary as I need to know wether the response is empty or not. For the getRandomSubmission the API Endpoint of Artemis returns a 200 HTTP Status code even though there might be no submission to be sent.

Reply:
There is a possibility to test for data.isEmpty. However it would also make most sense to throw an exception after this. So I think should be fine like this.

Reply:
Only way would be to rework the return type but this would mean a lot of refactoring and no real benefit